### PR TITLE
Fix analyze --watch command iterator

### LINKED ISSUE #96251: "flutter analyze --watch" seems to always crash on startup
## Command
```
flutter analyze --watch --write /tmp/analysis.devtools
```

## Steps to Reproduce
1. I ran the command above in root of the `devtools` repo.

## Logs
ConcurrentModificationError: (#0      List.addAll (dart:core-patch/growable_array.dart:325:11))
```
#0      List.addAll (dart:core-patch/growable_array.dart:325:11)
#1      AnalyzeContinuously._handleAnalysisStatus.<anonymous closure> (package:flutter_tools/src/commands/analyze_continuously.dart:116:18)
#2      _LinkedHashMapMixin.forEach (dart:collection-patch/compact_hash.dart:539:8)
#3      AnalyzeContinuously._handleAnalysisStatus (package:flutter_tools/src/commands/analyze_continuously.dart:114:22)
#4      AnalyzeContinuously.analyze.<anonymous closure> (package:flutter_tools/src/commands/analyze_continuously.dart:79:53)
#5      _rootRunUnary (dart:async/zone.dart:1434:47)
#6      _CustomZone.runUnary (dart:async/zone.dart:1335:19)
#7      _CustomZone.runUnaryGuarded (dart:async/zone.dart:1244:7)
#8      _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:341:11)
#9      _DelayedData.perform (dart:async/stream_impl.dart:591:14)
#10     _StreamImplEvents.handleNext (dart:async/stream_impl.dart:706:11)
#11     _PendingEvents.schedule.<anonymous closure> (dart:async/stream_impl.dart:663:7)
#12     _rootRun (dart:async/zone.dart:1418:47)
#13     _CustomZone.run (dart:async/zone.dart:1328:19)
#14     _CustomZone.runGuarded (dart:async/zone.dart:1236:7)
#15     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1276:23)
#16     _rootRun (dart:async/zone.dart:1426:13)
#17     _CustomZone.run (dart:async/zone.dart:1328:19)
#18     _CustomZone.runGuarded (dart:async/zone.dart:1236:7)
#19     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1276:23)
#20     _microtaskLoop (dart:async/schedule_microtask.dart:40:21)
#21     _startMicrotaskLoop (dart:async/schedule_microtask.dart:49:5)
#22     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:122:13)
#23     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:193:5)
```
```
[✓] Flutter (Channel master, 2.9.0-1.0.pre.266, on Ubuntu 20.04.3 LTS 5.4.0-91-generic, locale C.UTF-8)
    • Flutter version 2.9.0-1.0.pre.266 at /home/ianh/dev/flutter
    • Upstream repository git@github.com:flutter/flutter.git
    • Framework revision 4a66bfec7c (37 minutes ago), 2022-01-06 15:08:26 -0800
    • Engine revision 38a007d125
    • Dart version 2.16.0 (build 2.16.0-157.0.dev)
    • DevTools version 2.9.2
    • PATH =
      /home/ianh/Android/Sdk/platform-tools:/home/ianh/dev/flutter/bin:/home/ianh/dev/flutter/bin/cache/dart-sdk/bin/:/home/ianh/dev/usa
      ble-git:/home/ianh/.pub-cache/bin:/home/ianh/bin:/home/ianh/.nvm/versions/node/v14.17.3/bin:/home/ianh/Android/Sdk/platform-tools:
      /home/ianh/dev/flutter/bin:/home/ianh/dev/flutter/bin/cache/dart-sdk/bin/:/home/ianh/dev/usable-git:/home/ianh/.pub-cache/bin:/usr
      /local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/home/ianh/dev/depot_tools:/home/ia
      nh/dev/fuchsia/.jiri_root/bin:/home/ianh/bin/gh_2.0.0_linux_amd64/bin:/home/ianh/.rvm/bin:/home/ianh/.rvm/bin:/home/ianh/dev/depot
      _tools:/home/ianh/dev/fuchsia/.jiri_root/bin:/home/ianh/bin/gh_2.0.0_linux_amd64/bin:/home/ianh/.rvm/bin

[!] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /home/ianh/Android/Sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/linux#android-setup for more details.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Linux toolchain - develop for Linux desktop
    • clang version 10.0.0-4ubuntu1
    • cmake version 3.16.3
    • ninja version 1.8.2
    • pkg-config version 0.29.1

[✓] Android Studio (version 2020.3)
    • Android Studio at /home/ianh/bin/android-studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] Connected device (2 available)
    • Linux (desktop) • linux  • linux-x64      • Ubuntu 20.04.3 LTS 5.4.0-91-generic
    • Chrome (web)    • chrome • web-javascript • Google Chrome 96.0.4664.45

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.

```

## Flutter Application Metadata
No pubspec in working directory.

### PR DESCRIPTION
**Pull request by jmagman**
Avoid modifying the iterator of analyzer errors inside its `forEach` block. Fix shadowed variable.

Caused by https://github.com/flutter/flutter/pull/95442.
Fixes https://github.com/flutter/flutter/issues/96251.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter_tools/test/commands.shard/hermetic/analyze_continuously_test.dart [hunk 1]
```diff
@@ -228,14 +233,73 @@ void main() {
...
+      fileSystem: fileSystem,
+      processManager: processManager,
+    );
+
+    final TestFlutterCommandRunner commandRunner = TestFlutterCommandRunner();
+    commandRunner.addCommand(command);
+    unawaited(commandRunner.run(<String>['analyze', '--watch']));
+
+    Timer.periodic(const Duration(milliseconds: 10), (Timer timer) {
+      if (logger.statusText.contains('analyzed 1 file')) {
```
Comment (jmagman) [inline packages/flutter_tools/test/commands.shard/hermetic/analyze_continuously_test.dart:259]: I couldn't think of a better way to wait for the `AnalysisServer` to finish processing the errors other than spinning and checking stdout for the completion log.  Open to suggestions.

Comment (zanderso) [reply packages/flutter_tools/test/commands.shard/hermetic/analyze_continuously_test.dart:259]: Can you use `FakeAsync` and check after each pump of the message loop.

Comment (jmagman) [reply packages/flutter_tools/test/commands.shard/hermetic/analyze_continuously_test.dart:259]: Oh that's way better.

File: packages/flutter_tools/test/commands.shard/hermetic/analyze_continuously_test.dart [hunk 2]
```diff
@@ -174,7 +175,6 @@ void main() {
   });
 
   testUsingContext('Can run AnalysisService with customized cache location', () async {
-    final Completer<void> completer = Completer<void>();
```
Comment (jmagman) [inline packages/flutter_tools/test/commands.shard/hermetic/analyze_continuously_test.dart:177]: This completer wasn't actually doing anything.

**Review by zanderso — APPROVED** [review]
LGTM

**Pull request merged**